src/libostree/ostree-repo-commit.c

```diff
@@ -2268,6 +2268,179 @@ ostree_repo_write_commit_detached_metadata (OstreeRepo      *self,
   return TRUE;
 }
 
+gboolean
+_ostree_repo_read_commit_compat_signature (OstreeRepo      *self,
+                                           const char      *checksum,
+                                           GBytes         **out_signature,
+                                           GCancellable    *cancellable,
+                                           GError         **error)
+{
+  gboolean ret = FALSE;
+  char signature_path[_OSTREE_LOOSE_PATH_MAX];
+  g_autoptr(GBytes) ret_signature = NULL;
+  GError *temp_error = NULL;
+
+  _ostree_loose_path_with_extension (signature_path, checksum, "sig");
+  ret_signature = ot_file_mapat_bytes (self->objects_dir_fd,
+                                       signature_path, &temp_error);
+  if (!ret_signature)
+    {
+      if (g_error_matches (temp_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+        {
+          g_clear_error (&temp_error);
+        }
+      else
+        {
+          g_propagate_error (error, temp_error);
+          g_prefix_error (error, "Unable to read existing compat signature: ");
+          goto out;
+        }
+    }
+
+  ret = TRUE;
+  ot_transfer_out_value (out_signature, &ret_signature);
+ out:
+  return ret;
+}
+
+gboolean
+_ostree_repo_write_commit_compat_signature (OstreeRepo      *self,
+                                            const char      *checksum,
+                                            GBytes          *signature,
+                                            GCancellable    *cancellable,
+                                            GError         **error)
+{
+  gboolean ret = FALSE;
+  char signature_path[_OSTREE_LOOSE_PATH_MAX];
+  gsize signature_size;
+  gconstpointer signature_buf = NULL;
+
+  if (!_ostree_repo_ensure_loose_objdir_at (self->objects_dir_fd, checksum,
+                                            cancellable, error))
+    goto out;
```

Comment:
Optional: I think you could use early returns instead of `goto` (also for the other method below)